test/functional/feature_assumeutxo.py

```diff
@@ -248,6 +258,69 @@ def test_snapshot_not_on_most_work_chain(self, dump_output_path):
         node1.submitheader(main_block1)
         node1.submitheader(main_block2)
 
+    def test_sync_from_assumeutxo_node(self, snapshot):
+        """
+        This test verifies that:
+        1. An IBD node can sync headers from an AssumeUTXO node at any time.
+        2. IBD nodes do not request historical blocks from AssumeUTXO nodes while they are syncing the background-chain.
+        3. The assumeUTXO node dynamically adjusts the network services it offers according to its state.
+        4. IBD nodes can fully sync from AssumeUTXO nodes after they finish the background-chain sync.
+        """
+        self.log.info("Testing IBD-sync from assumeUTXO node")
+        # Node2 starts clean and loads the snapshot.
+        # Node3 starts clean and seeks to sync-up from snapshot_node.
+        miner = self.nodes[0]
+        snapshot_node = self.nodes[2]
+        ibd_node = self.nodes[3]
+
+        # Start test fresh by cleaning up node directories
```

Comment:
> If this is self-contained and you need to wipe everything this could have been done in a separate test line `p2p_assumeutxo.py` for example and you copy over the setup from this file here but it seems like there isn't much needed.

I started there (see me crying about the `feature_assumeutxo.py`  code organization here https://github.com/bitcoin/bitcoin/pull/30807#pullrequestreview-2281468112), but ended up moving it inside this all-in-one file because I wasn't happy copy pasting the costly snapshot setup into another test file (this test case doesn't need all of that).
I think we should allow assumeUTXO snapshot customization on regtest. It would make testing much simpler and self-contained. We would be able to subdivide it on network/p2p, local node, and wallet test files.
But still, if you don't mind, let's leave this convo for a follow-up so we can fix this bug for v28. I would be happy to tackle if we reach consensus over it.

Reply:
> I think we should allow assumeUTXO snapshot customization on regtest. It would make testing much simpler and self-contained.

How would it make it simpler? It's another thing to keep track of in the test code and the implementation code. If we need more snapshot heights we can always add them but I haven't felt the need for it so far when I have added new coverage.

Reply:
> > I think we should allow assumeUTXO snapshot customization on regtest. It would make testing much simpler and self-contained.
> 
> How would it make it simpler? It's another thing to keep track of in the test code and the implementation code.

The custom snapshot feature shouldn't take more than few lines at `init.cpp`. And for the test side, it would be an extra argument during init. I believe we will sooner or later need different snapshot for further wallet tests, maybe more than one in order to cover different self-contained test scenarios. I think it worth splitting `feature_assumeutxo.py` in `rpc_assumeutxo.py`, `p2p_assumeutxo.py`, `wallet_assumeutxo.py`, `validation_assumeutxo.py` etc.. for maintenance reasons.

>  If we need more snapshot heights we can always add them but I haven't felt the need for it so far when I have added new coverage.

This is simpler in the sense of not requiring to add new hardcoded values at chainparams every time we need something different. For example, this p2p test case just need a snapshot, it can be a very simple one. It does not require all the extra txs setup time.